crates/oxc_semantic/src/scope/builder.rs

```diff
@@ -57,6 +57,22 @@ impl ScopeBuilder {
         }
     }
 
+    pub fn resolve_reference(&mut self, symbol_table: &SymbolTable) {
+        let current_scope = self.current_scope();
+        for (variable, reference) in &current_scope.unresolved_references {
```

Comment:
You don't need the `RefCell` on `reference.resolved_symbol_id` if you drain and collect these on the stack and then mutate the references. 

The code is pretty hard to write here ...

Reply:
> You don't need the `RefCell` on `reference.resolved_symbol_id` if you drain and collect these on the stack and then mutate the references.

Done.